hyperstart.go

```diff
@@ -510,6 +509,23 @@ func (h *hyper) startPauseContainer(podID string) error {
 }
 
 func (h *hyper) startOneContainer(pod Pod, c Container) error {
+
+	// flag to determine if shim must be stopped
+	workloadStarted := false
+
+	//In case cannot start the container, stop the shim associated with it
+	defer func() {
+		//NOTE: workloadStarted must be false in case shim being stopped
+		if workloadStarted == true {
+			return
+		}
+
+		virtLog.Debug("failed start container, stoping shim")
+		if err := stopShim(c.process.Pid); err != nil {
```

Comment:
So the logic makes perfect sense and it's definitely a case we're not handling properly.
However, I would like it to be handled at a higher level, i.e. not at the hyperstart one, because this logic applies to all agent: When we fail to start a container, we should remove its shim.

So what I'm saying is that the shim deletion/stopping should be done from the `pod.go` and `container.go` files. We almost do that from `container.go`:

```Go
	err = c.pod.agent.startContainer(*(c.pod), *c)
	if err != nil {
		c.stop()
		return err
	}
```

But we don't for `pod.go`. `pod.stop()` should be called and it should take care of cleaning any dangling shim.